### PR TITLE
Allow tests to run using a list of python versions and refactoring test trigger config

### PR DESCRIPTION
**Pull request by avagraw**
---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to the IoT extension!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

Intent for Production

- [ ] It is expected that pull requests made to default or core branches such as `dev` or `main` are of production grade. Corollary to this, any merged contributions to these branches may be deployed in a public release at any given time. By checking this box, you agree and commit to the expected production quality of code.

Basic expectations

- [ ] If introducing new functionality or modified behavior, are they backed by unit and/or integration tests?
- [ ] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [ ] Have **all** the relevant unit **and** integration tests pass? i.e. `pytest <project root> -vv`. Please provide evidence in the form of a screenshot showing a succesful run of tests locally OR a link to a test pipeline that has been run against the change-set.
- [ ] Have linter checks passed using the `.pylintrc` and `.flake8` rules? Look at the CI scripts for example usage.
- [ ] Have extraneous print or debug statements, commented out code-blocks or code-statements (if any) been removed from the surface area of changes?
- [ ] Have you made an entry in HISTORY.rst which concisely explains your user-facing feature or change?

Azure IoT CLI maintainers reserve the right to enforce any of the outlined expectations.

A PR is considered **ready for review** when all basic expectations have been met (or do not apply).

### CODE REVIEW

File: .azure-devops/nightly.yml [hunk 1]
```diff
@@ -100,38 +108,41 @@ stages:
...
+    - stage: 'test_msi_${{python_version}}'
+      displayName: 'Run all tests against Windows MSI install on python ${{python_version}}'
+      pool:
+        vmImage: 'windows-2019'
+      dependsOn: test_min_${{python_version}}
+      jobs:
+      - template: templates/trigger-tests.yml
+        parameters:
+          azureCLIVersion: msi
+          pythonVersion: ${{replace(python_version, '_', '.')}}
```
Comment (digimaun) [inline .azure-devops/nightly.yml:137]: the MSI only has one python version

File: .azure-devops/create-release.yml [hunk 2]
```diff
@@ -21,14 +21,13 @@ parameters:
...
-- name: pythonVersionTestRun
-  type: string
-  default: '3.8.x'
-  values:
-  - '3.6.x'
-  - '3.8.x'
-  - '3.9.x'
-  - '3.10.x'
+- name: pythonVersionsForTesting
+  type: object
```
Comment (digimaun) [inline .azure-devops/create-release.yml:25]: What are the advantages of using `type: object` here?

Comment (avagraw) [reply .azure-devops/create-release.yml:25]: The type 'object' allows us to pass a list of python versions for which we want to run tests

File: .azure-devops/nightly.yml (hunk 1 above)
Comment (avagraw) [reply .azure-devops/nightly.yml:137]: Which python version is supported by MSI?

**Comment by avagraw** [comment]
Testing evidence: https://dev.azure.com/azureiotdevxp/aziotcli/_build/results?buildId=4180&view=logs&j=5542d1c7-9e73-545b-75f4-ad453094b234&t=9d46ab4d-242c-5e61-64ed-bafd5dafa3b2

File: .azure-devops/nightly.yml (hunk 1 above)
Comment (digimaun) [reply .azure-devops/nightly.yml:137]: It changes over time. The python runtime is loaded in the MSI, to be independent from system python runtimes. I believe more recently it provides a 3.8.x python version. In the past for a significant period its been 3.6.x. We should use the MSI python environment for tests - this is a key aspect for the MSI test.

**Comment by avagraw** [comment]
Release testing evidence: https://dev.azure.com/azureiotdevxp/aziotcli/_build/results?buildId=4295&view=logs&j=b965e18a-779e-5a51-69f7-83f4f4e9bf18&t=fe848c20-705f-5129-4392-2d280b1950c1

Nightly testing evidence: https://dev.azure.com/azureiotdevxp/aziotcli/_build/results?buildId=4302&view=logs&j=544d9072-97e4-5f50-bfdd-0240b22cf779

File: .azure-devops/templates/trigger-tests.yml [hunk 3]
```diff
@@ -33,64 +30,82 @@ jobs:
 - job: 'testCentral'
   displayName: 'Test IoT Central'
   condition: and(succeeded(), eq('${{ parameters.testCentral }}', true))
+  strategy:
+    matrix: $[ variables['pythonVersions'] ]
+    maxParallel: 2
```
Comment (digimaun) [inline .azure-devops/templates/trigger-tests.yml:35]: Any technology areas that do not provision the target resource for test must have a maxParallel of 1.

File: .azure-devops/nightly.yml [hunk 4]
```diff
@@ -99,29 +115,30 @@ stages:
           architecture: ${{ parameters.architecture }}
 
       - template: templates/install-and-record-version.yml
-
+
```
Comment (digimaun) [inline .azure-devops/nightly.yml:118]: Unneeded spaces?

**Comment by avagraw** [comment]
Release testing evidence: https://dev.azure.com/azureiotdevxp/aziotcli/_build/results?buildId=4310&view=logs&j=1f792e56-71fb-557a-81d8-19f62a947df4&t=1f792e56-71fb-557a-81d8-19f62a947df4

Nightly testing evidence: https://dev.azure.com/azureiotdevxp/aziotcli/_build/results?buildId=4308&view=logs&j=8e3ebb20-6c23-548d-5240-e229405aad81

File: .azure-devops/templates/trigger-tests.yml (hunk 3 above)
Comment (avagraw) [reply .azure-devops/templates/trigger-tests.yml]: Done

File: .azure-devops/nightly.yml (hunk 4 above)
Comment (avagraw) [reply .azure-devops/nightly.yml]: Fixed

**Comment by avagraw** [comment]
Release testing evidence: https://dev.azure.com/azureiotdevxp/aziotcli/_build/results?buildId=4338&view=logs&s=859b8d9a-8fd6-5a5c-6f5e-f84f1990894e

Nightly testing evidence: https://dev.azure.com/azureiotdevxp/aziotcli/_build/results?buildId=4339&view=logs&s=706c4132-ca96-5c63-cf02-edb0e3a7a479

**Comment by avagraw** [comment]
Central tests pass: https://dev.azure.com/azureiotdevxp/aziotcli/_build/results?buildId=4357&view=logs&j=8157547d-ccd1-59de-53d8-0ef338b7a610

**Review by digimaun — APPROVED** [review]

**Pull request merged**